src/api/services/goals.service.js

```diff
@@ -90,17 +90,15 @@ exports.deleteAll = async (goalId, users) => {
 exports.modifyMainGoal = async (mainGoalId, title) => {
   return await MainGoal.findByIdAndUpdate(mainGoalId, {
     $set: { title },
-  }).exec();
+  }).lean();
 };
 
-exports.modifySubGoal = async (subGoal, subGoalId, mainGoalId) => {
-  const { title, todos } = subGoal;
-  const mainGoal = await MainGoal.findById(mainGoalId).lean();
+exports.modifySubGoal = async (title, mainGoalId, subGoalId) => {
+  const mainGoal = await MainGoal.findById(mainGoalId).exec();
 
   mainGoal.subGoals.forEach(subGoal => {
     if (subGoalId === String(subGoal._id)) {
       subGoal.title = title;
```

Comment:
이전에 제가 작성했던 로직인데 ObjectId를 string으로 형변환을 하기 위해선 toString으로 쓰는 것이 알맞는다 하여 해당 부분 확인 후 수정해주시면 감사합니다🙏
```suggestion
    if (subGoalId === (subGoal._id.toString())) {
      subGoal.title = title;
```

Reply:
pull하면서 이력이 꼬였나보네요. 감사합니다.